snappy/click.go

```diff
@@ -176,6 +182,47 @@ ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.Target}} "$@"
 	return templateOut.String(), nil
 }
 
+func generateSnapDesktopFile(app *AppYaml, pkgPath string, m *snapYaml) (string, error) {
+	desktopTemplate := `[Desktop Entry]
+Encoding=UTF-8
+Type=Application
+Name={{.Name}}
+Comment={{.Comment}}
+Icon={{.Icon}}
+Exec={{.Exec}}{{if .Categories}}
+Categories={{range .Categories}}{{.}};{{end}}{{end}}
+`
+
+	// no desktop file needed
+	if app.Icon == "" && app.Comment == "" {
```

Comment:
There shouldn't be an app.Icon anymore after our agreements in Brazil, and Comment also sounds slightly obscure (comments are free form, meaningless). Let's please catch up online about this logic.